src/boot/bootctl-install.c

```diff
@@ -391,6 +391,8 @@ static int install_binaries(const char *esp_path, const char *arch, bool force)
         /* If we had a root directory to try, we didn't find it and we are in auto mode, retry on the host */
         if (r == -ENOENT && root && arg_install_source == ARG_INSTALL_SOURCE_AUTO)
                 r = chase_and_opendir(BOOTLIBDIR, NULL, CHASE_PREFIX_ROOT|CHASE_PROHIBIT_SYMLINKS, &path, &d);
+        if (r == -ENOENT && arg_graceful)
+                return log_debug_errno(r, "Boot loader directory does not exist, ignoring.");
```

Comment:
hmm? this is not graceful, you still propagate the error here.

But what's the goal here? that if the boot loader source doesn't exist at all we just don't install a boot loader?

maybe at least say "source directory", otherwise it's not clear which dir you actually mean